sycl/include/CL/sycl/half_type.hpp

```diff
@@ -27,9 +27,117 @@
 #define __SYCL_CONSTEXPR_ON_DEVICE
 #endif
 
+#ifdef _MSC_VER
```

Comment:
Please note that `clang` mimics other compilers ([docs](https://clang.llvm.org/docs/UsersManual.html#microsoft-extensions)) and therefore we will have this define present even when we compile this code by `clang`. 

If you really need this built-in, then I suggest to use [`__has_builtin`](https://clang.llvm.org/docs/LanguageExtensions.html#has-builtin) with corresponding macro guards to check for a presence of a built-in

Reply:
Changed that